spock-specs/src/test/groovy/org/spockframework/mock/runtime/ByteBuddyMockMakerSpec.groovy

```diff
@@ -0,0 +1,24 @@
+package org.spockframework.mock.runtime
+
+import spock.lang.Specification
+import spock.mock.MockMakerId
+
+import java.lang.reflect.Proxy
+
+class ByteBuddyMockMakerSpec extends Specification {
+
+  def "Verify ID"() {
+    expect:
+    MockMakerId.byteBuddy == "byte-buddy-mock-maker"
+  }
+
+  def "Use specific MockMaker byteBuddy"() {
+    when:
+    def m = Mock(mockMaker: MockMakerId.byteBuddy, Runnable)
+    def mockClass = m.getClass()
+    then:
+    !Proxy.isProxyClass(mockClass)
+    mockClass.name.contains('$SpockMock$')
+  }
```

Comment:
Add a test either here or on the `DetachedMockFactorySpec` to test if the selection works there as well.

Reply:
Done.